### PR TITLE
Auto scaling groups data model refactor

### PR DESCRIPTION
**Pull request by serge-wq**
### Summary

Migrating the ASG sync to use the data model. Nothing else was changed in the sync aside from creating the model and using it during the sync. 

### Checklist

Provide proof that this works (this makes reviews move faster). Please perform one or more of the following:
- [x] Update/add unit or integration tests.
- [ ] Include a screenshot showing what the graph looked like before and after your changes.
- [ ] Include console log trace showing what happened before and after your changes.

If you are changing a node or relationship:
- [ ] Update the [schema](https://github.com/lyft/cartography/tree/master/docs/root/modules) and [readme](https://github.com/lyft/cartography/blob/master/docs/schema/README.md).

If you are implementing a new intel module:
- [ ] Use the NodeSchema [data model](https://cartography-cncf.github.io/cartography/dev/writing-intel-modules.html#defining-a-node).

### CODE REVIEW

**Review by heryxpc — APPROVED** [review]

File: cartography/models/aws/ec2/auto_scaling_groups.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+@dataclass(frozen=True)
+class EC2InstanceAutoScalingGroupProperties(CartographyNodeProperties):
+    id: PropertyRef = PropertyRef('InstanceId')
+    instanceid: PropertyRef = PropertyRef('InstanceId', extra_index=True)
+    lastupdated: PropertyRef = PropertyRef('lastupdated', set_in_kwargs=True)
+    region: PropertyRef = PropertyRef('Region', set_in_kwargs=True)
+
+
+@dataclass(frozen=True)
+class EC2InstanceAutoScalingGroupSchema(CartographyNodeSchema):
```
Comment (achantavy) [inline cartography/models/aws/ec2/auto_scaling_groups.py:80]: Thank you for accounting for this

File: cartography/models/aws/ec2/launch_configurations.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    spot_price: PropertyRef = PropertyRef('SpotPrice')
+    iam_instance_profile: PropertyRef = PropertyRef('IamInstanceProfile')
+    ebs_optimized: PropertyRef = PropertyRef('EbsOptimized')
+    associate_public_ip_address: PropertyRef = PropertyRef('AssociatePublicIpAddress')
+    placement_tenancy: PropertyRef = PropertyRef('PlacementTenancy')
+    region: PropertyRef = PropertyRef('Region', set_in_kwargs=True)
+
+
+@dataclass(frozen=True)
+class LaunchConfiguratiionToAwsAccountRelProperties(CartographyRelProperties):
```
Comment (achantavy) [inline cartography/models/aws/ec2/launch_configurations.py:36]:
```suggestion
class LaunchConfigurationToAwsAccountRelProperties(CartographyRelProperties):
```

File: cartography/intel/aws/ec2/auto_scaling_groups.py [hunk 3]
```diff
@@ -1,243 +1,205 @@
...
+        AWS_ID=current_aws_account_id,
+        lastupdated=update_tag,
     )
-    neo4j_session.run(
-        ingest_lts,
-        autoscaling_groups_list=launch_templates,
-        AWS_ACCOUNT_ID=current_aws_account_id,
+
+
+def load_instances(
```
Comment (achantavy) [inline cartography/intel/aws/ec2/auto_scaling_groups.py:163]:
```suggestion
def load_instance_asgs(
```

Or similar. Just something to indicate that this is an instance as known by an ASG

File: cartography/intel/aws/ec2/auto_scaling_groups.py [hunk 4]
```diff
@@ -1,243 +1,205 @@
...
+        AWS_ID=current_aws_account_id,
+        lastupdated=update_tag,
     )
-    neo4j_session.run(
-        ingest_lcs,
-        autoscaling_groups_list=launch_configs,
-        AWS_ACCOUNT_ID=current_aws_account_id,
+
+
+def load_vpcs(
```
Comment (achantavy) [inline cartography/intel/aws/ec2/auto_scaling_groups.py:150]:
```suggestion
def load_subnet_asg(
```

Or similar

File: tests/integration/cartography/intel/aws/ec2/test_ec2_auto_scaling_groups.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        node_1_attr='id',
+        node_2_label='LaunchConfiguration',
+        node_2_attr='name',
+        rel_label='HAS_LAUNCH_CONFIG',
+        rel_direction_right=True,
+    ) == {
+        (GET_AUTO_SCALING_GROUPS[0]['AutoScalingGroupARN'], GET_LAUNCH_CONFIGURATIONS[0]['LaunchConfigurationName']),
+    }
+
+    assert check_rels(
```
Comment (achantavy) [inline tests/integration/cartography/intel/aws/ec2/test_ec2_auto_scaling_groups.py:84]: Could we add one more test for the ASG-to-AWS account edge?

**Review by achantavy — APPROVED** [review]
Thank you for taking this on! It looks good to me, just caught a typo and left some naming nits

File: tests/integration/cartography/intel/aws/ec2/test_ec2_auto_scaling_groups.py (hunk 5 above)
Comment (serge-wq) [reply tests/integration/cartography/intel/aws/ec2/test_ec2_auto_scaling_groups.py:99]: Sure.

File: cartography/intel/aws/ec2/auto_scaling_groups.py (hunk 4 above)
Comment (serge-wq) [reply cartography/intel/aws/ec2/auto_scaling_groups.py]: Updated both names to `load_asg_[subnets|instances]`
